benchmark/Benchmark/Type.hs

```diff
@@ -1,73 +1,53 @@
 module Benchmark.Type
-       ( test
-       , testCodec
+       ( HaskellType(..)
+       , FruitInside(..)
+       , SizeInside(..)
        ) where
 
+import Data.Aeson.Types (FromJSON, parseJSON, withObject, (.:))
 import Data.Text (Text)
-import Data.Time (Day, LocalTime (..), TimeOfDay, ZonedTime)
-
-import Toml (TomlCodec, pretty, (.=))
-import Toml.Parser (parse)
-
-import qualified Data.Text.IO as TIO
-import qualified Toml
-
-
-test :: IO ()
-test = do
-    tomlFile <- TIO.readFile "./benchmark/benchmark.toml"
-    TIO.putStrLn $ case Toml.decode testCodec tomlFile of
-        Left msg    -> Toml.prettyException msg
-        Right value -> Toml.encode testCodec value
-    case parse tomlFile of
-        Left (Toml.ParseException msg) -> TIO.putStrLn msg
-        Right toml -> TIO.putStrLn $ case Toml.runTomlCodec testCodec toml of
-            Left msg    -> Toml.prettyException msg
-            Right value -> pretty $ Toml.execTomlCodec testCodec value
-
-data Test = Test
-    { testTitle  :: Text
-    , testAtom   :: Double
-    , testCash   :: Bool
-    , testWords  :: [Text]
-    , testDigits :: [Double]
-    , testBool   :: [Bool]
-    , testDate   :: [Day]
-    , testToday  :: ZonedTime
-    , testFruit  :: FruitInside
-    , testSize   :: SizeInside
+import Data.Time (ZonedTime)
+
+
+-- | Haskell type to convert to.
+data HaskellType = HaskellType
+    { htTitle :: Text
+    , htAtom  :: Double
+    , htCash  :: Bool
+    , htWords :: [Text]
+    , htBool  :: [Bool]
+    , htToday :: ZonedTime
+    , htFruit :: FruitInside
+    , htSize  :: SizeInside
     }
+    deriving Show
+
+instance FromJSON HaskellType where
+    parseJSON = withObject "HaskellType" $ \o -> HaskellType
+        <$> o .: "title"
+        <*> o .: "atom"
+        <*> o .: "cash"
+        <*> o .: "words"
+        <*> o .: "bool"
+        <*> o .: "today"
+        <*> o .: "fruit"
+        <*> o .: "size"
 
 data FruitInside = FruitInside
     { fiName        :: Text
     , fiDescription :: Text
-    , fiHarvest     :: LocalTime
-    , fiDue         :: TimeOfDay
     }
+    deriving Show
 
-insideF :: TomlCodec FruitInside
-insideF = FruitInside
-    <$> Toml.text "name" .= fiName
-    <*> Toml.text "description" .= fiDescription
-    <*> Toml.localTime "harvest" .= fiHarvest
-    <*> Toml.timeOfDay "due" .= fiDue
+instance FromJSON FruitInside where
+    parseJSON = withObject "FruitInside" $ \o -> FruitInside
+        <$> o .: "name"
+        <*> o .: "description"
 
 newtype SizeInside = SizeInside
     { unSize :: [[Double]]
     }
+    deriving Show
```

Comment:
According to our style-guide `deriving` should go on the same line as `}` after `}`